android/libraries/rib-base/src/main/java/com/badoo/ribs/android/recyclerview/RecyclerViewHost.kt

```diff
@@ -0,0 +1,53 @@
+package com.badoo.ribs.android.recyclerview
+
+import android.content.Context
+import android.os.Parcelable
+import android.widget.FrameLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.badoo.ribs.android.recyclerview.client.RecyclerViewRibResolver
+import com.badoo.ribs.core.Rib
+import io.reactivex.ObservableSource
+import kotlinx.android.parcel.Parcelize
+
+/**
+ * Considered experimental. Handle with care.
+ */
+interface RecyclerViewHost : Rib {
+
+    interface Dependency<T : Parcelable> {
+        fun hostingStrategy(): HostingStrategy
+        fun initialElements(): List<T>
+        fun recyclerViewHostInput(): ObservableSource<Input<T>>
+        fun resolver(): RecyclerViewRibResolver<T>
+        fun layoutManagerFactory(): LayoutManagerFactory
+        fun viewHolderLayoutParams(): FrameLayout.LayoutParams
+
+        interface LayoutManagerFactory: (Context) -> RecyclerView.LayoutManager {
+            companion object {
+                val linear = object : LayoutManagerFactory {
```

Comment:
I think that we shouldn't use predefined const, better to provide fun which will create factory for you, like:
```
inline fun LayoutManagerFactory(crossinline builder: (Context) -> RecyclerView.LayoutManager) =
    object : LayoutManagerFactory{
        override fun invoke(context: Context): RecyclerView.LayoutManager =
            builder(context)
    }

fun howToUse() {
    LayoutManagerFactory(::LinearLayoutManager)
}
```

Reply:
inline will not compile unfortunately, but leaving it a typealias instead of an interface will, and it will work even better in client code: 
```= ::LinearLayoutManager```

thanks!